scim/groupRecords.bal

```diff
@@ -0,0 +1,132 @@
+    string displayName;
+    Member[] members?;
+    Meta meta?;
+    Role roles?;
+};
+
+# Represents the SCIM Group resource request body used for createGroup
+#
+# + schemas - URIs of the used SCIM schemas  
+# + displayName - Name for the Group  
+# + members - List of members of the Group  
+# + meta - Metadata of the group  
+# + roles - List of roles for the group
+public type GroupCreate record {
+    string[] schemas;
+    string displayName;
+    Member[] members?;
+    Meta meta?;
+    Role roles?;
+};
+
+# Represents the sub-attributes of members attribute of SCIM Group resource schema 
+#
+# + value - Value of an "id" attribute of a SCIM resource  
+# + display - Name used to display 
+# + \$ref - Reference URI
+public type Member record {
+    string value?;
+    string display?;
+    string \$ref?;
+};
+
+# Represents the response of by getGroups and searchGroup methods
+#
+# + totalResults - Total number of results  
+# + startIndex - Starting index of the returned results  
+# + itemsPerPage - Number of results returned per page  
+# + schemas - The list of schema URIs used  
+# + Resources - The list of group resources
+public type GroupResponse record {
+    int totalResults?;
+    int startIndex?;
+    int itemsPerPage?;
+    string[] schemas?;
+    GroupResource[] Resources?;
+};
+
+# Represents the Response of createGroup and getGroup methods
+#
+# + displayName - Name for the Group  
+# + meta - Metadata of the group
+# + members - List of members of the Group 
+# + roles - List of roles for the group 
+# + id - Unique identifier by service provider
+# + schemas - The list of schema URIs used
+public type GroupResource record {
+    string displayName?;
+    Meta meta;
+    Member[] members?;
+    Role[] roles?;
+    string id?;
+    string[] schemas?;
+};
+
+# Represents the Response of updateGroup method
+#
+# + schemas - The list of schema URIs used
+# + displayName - Name for the Group
+# + members - List of members of the Group 
+# + meta - Metadata of the group  
+# + roles - List of roles for the group
+public type GroupUpdate record {
+    string[] schemas?;
+    string displayName?;
+    Member[] members?;
+    Meta meta?;
+    Role roles?;
+};
+
+# Represents the Request body of patchGroup method
+#
+# + schemas - URIs of the used SCIM schemas  
+# + Operations - List of operations
+public type GroupPatch record {
+    string[] schemas;
+    PatchOpGroup[] Operations;
+};
+
+# Represents the sub-attributes of Operations attribute of GroupPatch record
+#
+# + op - Operation to be performed 
+# + value - Value to be used for the operation
+public type PatchOpGroup record {
+    string op;
+    GroupUpdate value;
+};
+
+# Represents the Request body of searchGroup method
+#
+# + schemas - URIs of the used SCIM schemas  
+# + filter - Filter expression 
+# + startIndex - Starting index of the returned results
+public type GroupSearch record {
+    string[] schemas;
+    string filter;
+    int startIndex?;
```

Comment:
https://www.rfc-editor.org/rfc/rfc7644#section-3.4.3
Any reason to miss the
attributes, excludedAttributes, filter, sortBy and sortOrder query params?

Reply:
In addition `domain` query param is supported for /Groups endpoint in WSO2 IS SCIM provider.
You have considered that in UserSearch record. If so, it should be valid to group search too